<launch>
<param name="use_sim_time" value="true"/>
<arg name="weights_path" default="$(find yolov5_pytorch_ros)/weights/tsudanuma_2-3_sim.pt"/>
<arg name="landmark_name_file"   default="$(find landmark_map_builder)/param/name.yaml"/>
<arg name="landmark_record_file" default="$(find landmark_map_builder)/landmark/landmark_ver4.yaml"/>

<node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" output="screen">
  <param name="map_update_interval" value="5.0"/>
  <param name="maxUrange" value="16.0"/>
  <param name="sigma" value="0.05"/>
  <param name="kernelSize" value="1"/>
  <param name="lstep" value="0.05"/>
  <param name="astep" value="0.05"/>
  <param name="iterations" value="5"/>
  <param name="lsigma" value="0.075"/>
  <param name="ogain" value="3.0"/>
  <param name="lskip" value="0"/>
  <param name="srr" value="0.1"/>
  <param name="srt" value="0.2"/>
  <param name="str" value="0.1"/>
  <param name="stt" value="0.2"/>
  <param name="linearUpdate" value="1.0"/>
  <param name="angularUpdate" value="0.5"/>
  <param name="temporalUpdate" value="3.0"/>
  <param name="resampleThreshold" value="0.5"/>
  <param name="particles" value="30"/>
  <param name="xmin" value="-50.0"/>
  <param name="ymin" value="-50.0"/>
  <param name="xmax" value="50.0"/>
  <param name="ymax" value="50.0"/>
  <param name="delta" value="0.05"/>
  <param name="llsamplerange" value="0.01"/>
  <param name="llsamplestep" value="0.01"/>
  <param name="lasamplerange" value="0.005"/>
  <param name="lasamplestep" value="0.005"/>
</node>

<include file="$(find yolov5_pytorch_ros)/launch/landmark_detector.launch">
  <arg name="weights_path" value="$(arg weights_path)"/>
</include>

<include file="$(find landmark_map_builder)launch/record_landmark.launch">
  <arg name="landmark_name_file"   value="$(arg landmark_name_file)"/>
  <arg name="landmark_record_file" value="$(arg landmark_record_file)"/>
</include>

<node pkg="rviz" type="rviz" name="rviz" required="true" args="-d $(find navigation)/rviz/mapping.rviz"/>
</launch>